packages/expo-enclave/android/src/main/java/expo/modules/enclave/KeyManager.kt

```diff
@@ -0,0 +1,155 @@
+package expo.modules.enclave
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import java.security.spec.ECGenParameterSpec
+import java.security.KeyPairGenerator
+import java.security.KeyPair
+import java.security.KeyStore
+import java.security.KeyStoreException
+import java.security.Signature
+import org.bouncycastle.jce.ECPointUtil
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import java.util.concurrent.Executor
+import android.os.Bundle
+import androidx.fragment.app.FragmentActivity
+import android.content.Context
+import java.util.concurrent.Executors
+import expo.modules.core.ModuleRegistry
+import expo.modules.core.interfaces.ActivityProvider
+import android.app.Activity
+import expo.modules.core.interfaces.services.UIManager
+import expo.modules.core.Promise
+import android.os.Build
+import android.content.pm.PackageManager
+
+
+fun ByteArray.toHexString() = joinToString("") { "%02x".format(it) }
+fun String.decodeHex(): ByteArray {
+    check(length % 2 == 0) { "Must have an even length" }
+
+    return chunked(2)
+        .map { it.toInt(16).toByte() }
+        .toByteArray()
+}
+
+class SigningCallback(_message: String) {
```

Comment:
a nit, you might be able to remove this class and have a lambda down in `sign()` that resolves the promise